### PR TITLE
[Feature][Connector-V2][Klaviyo]Add Klaviyo source connector

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

https://github.com/apache/incubator-seatunnel/issues/3018
Add Klaviyo source connector

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TaoZex** [comment]
@hailin0 @EricJoy2048 @TyrantLucifer PTAL

File: docs/en/Connector-v2-release-state.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ SeaTunnel uses a grading system for connectors to help you understand what to ex
 | [IoTDB](connector-v2/sink/IoTDB.md)                         | Sink   | Beta   | 2.2.0-beta      |
 | [Jdbc](connector-v2/source/Jdbc.md)                         | Source | Beta   | 2.2.0-beta      |
 | [Jdbc](connector-v2/sink/Jdbc.md)                           | Sink   | Beta   | 2.2.0-beta      |
+| [Klaviyo](connector-v2/source/Klaviyo.md)                   | Source | Alpha  | 2.2.0-beta      |
```
Comment (CalvinKirs) [inline docs/en/Connector-v2-release-state.md]: We haven't released this yet.

File: docs/en/connector-v2/source/Klaviyo.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+##  Options
+
+| name                        | type   | required | default value |
+| --------------------------- | ------ | -------- | ------------- |
+| url                         | String | Yes      | -             |
+| private_key                 | String | Yes      | -             |
+| revision                    | String | Yes      | -             |
+| method                      | String | No       | get           |
+| schema.fields               | Config | No       | -             |
```
Comment (CalvinKirs) [inline docs/en/connector-v2/source/Klaviyo.md:27]: This field is usually defined as a `schema`

File: docs/en/Connector-v2-release-state.md (hunk 1 above)
Comment (TaoZex) [reply docs/en/Connector-v2-release-state.md]: Thanks.

File: docs/en/connector-v2/source/Klaviyo.md (hunk 2 above)
Comment (TaoZex) [reply docs/en/connector-v2/source/Klaviyo.md:27]: Thanks for your review.I have changed it to be the same as http source.

**Review by hailin0 — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**